NPE when updating neighbours

I just had this error on all my nodes during startup:
```
12:53:05.732 [WARN ] [ding.zwave.handler.ZWaveThingHandler] - NODE 5: Controller failed to register event handler.
12:53:05.735 [ERROR] [ome.core.thing.internal.ThingManager] - Exception occured during notification of thing 'zwave:fibaro_fgms001_00_000:ad1f4a49:node5' about bridge initialization at 'org.openhab.binding.zwave.handler.ZWaveThingHandler@c42d9d': null
java.lang.NullPointerException
	at org.openhab.binding.zwave.handler.ZWaveThingHandler.updateNeighbours(ZWaveThingHandler.java:924)[170:org.openhab.binding.zwave:2.0.0.201603010202]
	at org.openhab.binding.zwave.handler.ZWaveThingHandler.bridgeHandlerInitialized(ZWaveThingHandler.java:307)[170:org.openhab.binding.zwave:2.0.0.201603010202]
	at org.eclipse.smarthome.core.thing.internal.ThingManager$8.run(ThingManager.java:686)[98:org.eclipse.smarthome.core.thing:0.8.0.201602251230]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_65]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)[:1.8.0_65]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_65]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_65]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_65]
```

--- a/addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveControllerHandler.java
+++ b/addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveControllerHandler.java
@@ -119,8 +119,6 @@ public abstract class ZWaveControllerHandler extends BaseBridgeHandler implement
         // And register it as an OSGi service
         discoveryRegistration = bundleContext.registerService(DiscoveryService.class.getName(), discoveryService,
                 new Hashtable<String, Object>());
-
-        updateStatus(ThingStatus.INITIALIZING);
     }
 
     @Override

--- a/addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java
+++ b/addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java
@@ -295,11 +295,10 @@ public class ZWaveThingHandler extends BaseThingHandler implements ZWaveEventLis
         // This ensures we get called whenever there's an event we might be interested in
         if (((ZWaveControllerHandler) thingHandler).addEventListener(this) == true) {
             controllerHandler = (ZWaveControllerHandler) thingHandler;
+            updateNeighbours();
         } else {
             logger.warn("NODE {}: Controller failed to register event handler.", nodeId);
         }
-
-        updateNeighbours();
     }
 
     @Override
@@ -913,6 +912,10 @@ public class ZWaveThingHandler extends BaseThingHandler implements ZWaveEventLis
     }
 
     private void updateNeighbours() {
+        if (controllerHandler == null) {
+            return;
+        }
+
         ZWaveNode node = controllerHandler.getNode(nodeId);
         if (node == null) {
             return;

--- a/addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java
+++ b/addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/handler/ZWaveThingHandler.java
@@ -212,9 +212,6 @@ public class ZWaveThingHandler extends BaseThingHandler implements ZWaveEventLis
                 bridgeHandlerInitialized(handler, bridge);
                 // }
             }
-        } else {
-            // Until we get an update put the Thing into initialisation state
-            updateStatus(ThingStatus.INITIALIZING);
         }
 
         startPolling();
@@ -280,8 +277,6 @@ public class ZWaveThingHandler extends BaseThingHandler implements ZWaveEventLis
         } else {
             switch (node.getNodeState()) {
                 case INITIALIZING:
-                    updateStatus(ThingStatus.INITIALIZING);
-                    break;
                 case ALIVE:
                     updateStatus(ThingStatus.ONLINE);
                     break;
@@ -880,9 +875,6 @@ public class ZWaveThingHandler extends BaseThingHandler implements ZWaveEventLis
 
             switch (event.getState()) {
                 case INITIALIZING:
-                    logger.debug("NODE {}: Setting INITIALIZING", nodeId);
-                    updateStatus(ThingStatus.INITIALIZING);
-                    break;
                 case ALIVE:
                     logger.debug("NODE {}: Setting ONLINE", nodeId);
                     updateStatus(ThingStatus.ONLINE);
@@ -905,8 +897,8 @@ public class ZWaveThingHandler extends BaseThingHandler implements ZWaveEventLis
                     updateStatus(ThingStatus.ONLINE);
                     break;
                 default:
-                    logger.debug("NODE {}: Setting INITIALIZING: {}", nodeId, initEvent.getStage());
-                    updateStatus(ThingStatus.INITIALIZING, ThingStatusDetail.NONE, initEvent.getStage().toString());
+                    logger.debug("NODE {}: Setting ONLINE (INITIALIZING): {}", nodeId, initEvent.getStage());
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, initEvent.getStage().toString());
                     break;
             }
         }

